### ISSUE TITLE
When deserializing Option json4s fails silently on missing properties

### ISSUE THREAD

**Issue by valdemaras-wix** [json4s/json4s#356]
Here's an example
```
case class Application(user: Option[User])
case class User(firstName: String, lastName: String)   
implicit val formats = DefaultFormats
// note bellow there's no lastName in the json
val node = JsonMethods.parse("""{"user": {"firstName": "John"}}""")
val app = node.extract[Application]
```

Here instead of failing the deserialization of the user because of the missing lastName the app ends up as `Application(user = None)`

Any way to fail the deserialization instead of silently setting the whole user to None?

**Comment by raimohanska** [comment]
+1000 :) This causes me tons of headache right now

**Comment by raimohanska** [comment]
This is the solution:

   val formats: Formats =  new DefaultFormats {
     override val strictOptionParsing: Boolean = true
   }

Should be included in the documentation/readme IMHO

**Comment by valdemaras-wix** [comment]
Yep, I can confirm that the solution @raimohanska solution works -- took me some time to find it though.

I can second that it should be included into the documentation.
